src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -300,7 +204,7 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             // select last color scheme (avoid out of bounds error)
             ColorSchemeComboBox().SelectedIndex(removedSchemeIndex - 1);
         }
-        _ColorSchemeList.RemoveAt(removedSchemeIndex);
+        ViewModel().AllColorSchemes().RemoveAt(removedSchemeIndex);
```

Comment:
The view model should be removing the schemeVM from its own internal list, the view should not be reaching into the VM's list to remove it